cmake/podspec_cmake.rb

```diff
@@ -0,0 +1,494 @@
+#!/usr/bin/env ruby
```

Comment:
I've given this file a quick read, but I don't know ruby, so I've probably missed lots.

Also, we apparently have a ruby style guide: go/rubystyle. I haven't read it. If you haven't either, you might consider giving it a skim.

Something else to consider if you haven't already: A templating approach would be more natural here (though I suspect may not actually be possible due to at least the header mapping; though maybe templating + minor scripting?). i.e. create an "input" cmake file with `@VAR@`s for the things to be substituted and use sed (or cmake's configure_file, or m4 or any number of different options) to insert the relevant parts. You'd still need something to read/understand the podspec though, so this might not save much compared to doing it all via code.

Reply:
I considered this but abandoned it because CMake does not deal well with zero value cases. For example, `add_library(target STATIC)` is an error. So, either the templates need to add section only if there are values to substitute within the template or the CMake code within the template has to be aware of this. Either way it leaves the template or generated code far less comprehensible than what I've done here (I think).

All told I think the amount of code dedicated to the generated lexical structure is quite small so I think where this landed is reasonable. I'll be happy to revisit if the generated code gets more complex.

Reply:
I was already (mostly) following ruby-style, though I got doc comments for functions slightly wrong. Fixed.